modules/apps/object/object-web/src/main/resources/META-INF/resources/js/tests/components/ModalDeleteObjectEntry.spec.tsx

```diff
@@ -0,0 +1,89 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2025 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import '@testing-library/jest-dom/extend-expect';
+import {cleanup, render} from '@testing-library/react';
+
+import '@testing-library/jest-dom';
+import React from 'react';
+
+import ModalDeleteObjectEntry from '../../components/ModalDeleteObjectEntry';
+
+const MOCK_OBJECT_ENTRY = {
+	externalReferenceCode: 'test',
+	id: 1,
+	objectDefinitionId: 1,
+	objectEntry: null,
+};
+
+describe('ModalDeleteObjectEntry', () => {
+	const {Liferay: originalLiferay} = window;
+
+	beforeAll(() => {
+		window['Liferay'] = {
+			...originalLiferay,
+
+			// @ts-ignore
+
+			detach: (name, fn) => {
+
+				// @ts-ignore
+
+				window.removeEventListener(name, fn);
+			},
+			fire: (name, payload) => {
+				const event = document.createEvent('CustomEvent');
+
+				event.initCustomEvent(name);
+
+				if (payload) {
+					Object.keys(payload).forEach((key: string) => {
+
+						// @ts-ignore
+
+						event[key] = payload[key];
+					});
+				}
+
+				window.dispatchEvent(event);
+			},
+
+			// @ts-ignore
+
+			on: (name, fn) => {
+
+				// @ts-ignore
+
+				window.addEventListener(name, fn);
+			},
+		};
+	});
+
+	afterAll(() => {
+		cleanup();
+
+		window.Liferay = originalLiferay;
+
+		jest.resetAllMocks();
+	});
+
+	it('can disable the delete button after clicking on it', async () => {
+		const {findByText} = render(
+			<ModalDeleteObjectEntry byExternalReferenceCodePath="/test" />
+		);
+
+		Liferay.fire('openModalDeleteObjectEntry', {
+			objectEntry: MOCK_OBJECT_ENTRY,
+		});
+
+		const deleteButton = await findByText('delete');
+
+		expect(deleteButton).not.toBeDisabled();
+
+		deleteButton.click();
```

Comment:
Hi @DavysonMelo, I'm concerned about the absence of the userEvent. I found this nice explanation:

<img width="798" height="1113" alt="image" src="https://github.com/user-attachments/assets/b179280d-821e-44ea-bad8-1fd75b1a17bf" />